lib/moduleEncoding.js

```diff
@@ -262,23 +555,29 @@ async function compressModule(data, options) {
 		throw new RangeError('Invalid size of the module data');
 	}
 	let dataOffs = 0;
+	const prefixSize = prefix.prefixSize;
+	if (prefixSize < MODULE_PREFIX_SIZE) {
+		throw new RangeError('Invalid prefix data');
+	}
 	if (flags & ModuleFlags.DROP_MODULE_INFO) {
 		if (prefixOffs !== 0) {
 			throw new RangeError('DROP_MODULE_INFO can\'t be set for a module with a vector table');
 		}
 		// Skip the prefix data
-		dataOffs = MODULE_PREFIX_SIZE;
-		dataSize -= MODULE_PREFIX_SIZE;
+		dataOffs = prefixSize;
+		dataSize -= prefixSize;
+		// No need to keep suffix around within the compressed section either
+		dataSize -= suffixSize + 4 /* CRC-32 */;
```

Comment:
I think Device OS, at least old versions of it, expect the suffix data to be decompressed along with the rest of the module data. If so, this will break compressed OTA in Device Service for affected devices.